dapps/frenemies/src/routes/Setup.tsx

```diff
@@ -83,11 +103,18 @@ export function Setup() {
       };
 
       const serializer = new LocalTxnDataSerializer(provider);
-      const serializedTx = await serializer.serializeToBytes(currentAccount, submitTx);
-      const dryRunRes = await provider.dryRunTransaction(serializedTx.toString());
-
-      if (dryRunRes.status.status == 'failure') {
-        throw new Error(`Transaction would've failed with a reason '${dryRunRes.status.error}'`);
+      const serializedTx = await serializer.serializeToBytes(
+        currentAccount,
+        submitTx
+      );
+      const dryRunRes = await provider.dryRunTransaction(
+        serializedTx.toString()
+      );
+
+      if (dryRunRes.status.status == "failure") {
+        throw new Error(
+          `Transaction would've failed with a reason '${dryRunRes.status.error}'`
+        );
       }
 
       await signAndExecuteTransaction(submitTx);
```

Comment:
We have an issue here. For some reason takes some time to apply results on the fullnode and user can get stuck on the loading screen. Solution is to look at created objects / set user as registered if transaction went through (even tho "getScorecard" will return "null").